### ISSUE TITLE
Make Brave-specific options configurable through Windows Group Policy templates

### ISSUE THREAD

**Issue by mherrmann** [brave/brave-browser#26502]
## Test plan (added by @bsclifton)
1. Be on Windows
2. Windows + R, type in `gpedit.msc`, hit enter
3. Under `Computer Configuration`, right click `Administrative Templates`
4. In the context menu, pick `Add/Remove Templates...`
5. Download the templates from https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy (link should be https://brave-browser-downloads.s3.brave.com/latest/policy_templates.zip). We'll want to make sure these are the latest (check with @bsclifton or @mihaiplesa)
6. Unzip this `policy_templates.zip` and back in `gpedit.msc`, click `Add...`
7. Browse to the directory you just unzipped and find the `windows\adm\en-US\brave.adm` file
8. Select this file and pick `Open`
9. You should now see Brave under `Classic Administrative Templates`. In the `Brave Software` folder, you'll see the actual policies.
    ![Image](https://github.com/user-attachments/assets/408e10ee-f0b9-478b-b53d-be9db71e1866)
10. Go ahead and set this policy - change Tor disabled, etc.
11. Launch into Brave and verify that policies changed show in brave://policy/. You should also see `Managed by your organization` in the hamburger menu.
12. Policy description should have a link to take you to https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy
13. Verify the policy actually works

## Original issue description
This is a follow-up to #26501. There, we provide templates for changing general browser settings. The goal of this present issue is to build on that work and make it possible to configure Brave-specific browser options via Group Policy settings. Per the [current documentation](https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy), the minimal settings that should be supported are:

* TorDisabled
* IPFSEnabled
* BraveRewardsDisabled
* BraveWalletDisabled

Related:

 * #9458
 * #25394
 * #20431
 * #18012
 * #8907
 * #6291

**Comment by Emi-TheDhamphirInLoveWithTheFrozenGhost** [comment]
It is nice Brave implemented a policy for Up/Down Shields https://github.com/brave/brave-browser/issues/25394

@bsclifton @mherrmann @spylogsster 

I think it would be good if Brave expanded even more Shields/Adblock policies to make Brave a really good browser for admins to control what they want the organizations to see within Brave and even for Parental Control.

I see many people requesting a _block site_ feature, and people talk about DNS and all that way of 'blocking' websites, but technically Brave can already do that, it only needs ways to stop anyone from bypassing them easily when they are not meant to.

First, one important policy/feature would be to have a way for 'organization' to be able to set custom lists and rules that no user can delete or disable, just like how Windows Firewalls features and rules can be set through GPO.

Second, and somehow a more important feature/flag/policy, it's to stop users from clicking the proceed button if/when a domain is blocked by the adblocker or change the warning screen to reflect it was blocked by a organization or something. You know, the little screen which is done by using `||example^` or `$document` with the first party domain blocking feature https://github.com/brave/brave-browser/issues/14134
Example:
<img src="https://user-images.githubusercontent.com/387249/111383107-06de8f80-8665-11eb-8eb2-a637d76dbd45.png" height= 200px />

Just adding these two policies will add a way for organizations, admins, parents, schools to easily restrict/block websites without having to deal with another feature and/or Brave Team having to implement a feature that is already done with the adblocker, because the feature is already there, it only needs ways to prevent users to proceed or disable the rules.

Something important to note and why I say Brave already can block sites, it's that when Brave shows the warning screen, Brave hasn't connected to the Domain/IP, it shows the warning based on just on what the URL says before anything gets resolved. The website will only resolve and appear in DNS server logs or firewalls or anything when the user clicks the Proceed button, making it a great way (and already built) to block websites completely.
Otherwise if the user clicks proceed, everything gets blocked: scripts, fonts, images, spreadsheets, etc, (when `||example^` is used, not when just $document is used) but the main html document will load, which means Brave will be forced to make 1 connection, and since it will be plain html, then users can still read text which might not be good. That's why not allowing users to proceed is important, to make Brave adblocker a better tool for everything security and privacy.

If adblocker is used for this kind of stuff, not only it will be 'one feature less to build' by Brave Team, but also, no DNS server or VPN or anything else (but using another web browser) can bypass it for the way these document block works which is done only for what it is seen in the URL.

Which means, (another good point about using Adblocker for 'website blocker' feature) is that it can be done in flexible ways.

Example:

You want to block `example.com` but you want to allow people accessing `example.com/category/download/software` it can be done by using $document

`||example.com^$document`
and `@@||example.com/category/download/software` So it doesn't have the same limitations as DNS based blocking website features/programs/servers

And if we count the fact that the adblocker has scriptlet injections like window-close-if or CSP features like sandbox that can prevent popups and stops downloads in whatever website, using CSP is not perfect since people can middle click and download the file that way, but it can prevent automatic downloads, to truly stop downloads, again, $document for the way it works by searching in the URL, would be able to stop the download, so a rule like `/(.zip|.exe|.rar)/$document` will stop all downloads from happening anywhere if an organization or parent desires, making it superior to any available tool if done right, out of the box and without doing anything but using what it is already available.

I tried to find other issues with this type of request, so hope it is not duplicate, but hope this helps to understand how good these two policies can be.
Of course, many flags or policies can be done to control every aspect of the Adblocker but I think these two policies can be the most important to make available for any organization or parent to control websites they want their employees or children or students to have access.

Thank you and have a good day!.

**Comment by DesertBear** [comment]
The below setting also needs to be added, so that the VPN feature can be disabled across managed devices.

- BraveVPNDisabled

**Comment by bsclifton** [comment]
@DesertBear thanks for requesting that - I logged that one with https://github.com/brave/brave-browser/issues/29397 and we can look at this soon 😄

**Comment by rowansc1** [comment]
Hiya, 
Is there any further progress with the `TorDisabled` function being able to be edited via GPO? More specifically the progress of [16351](https://github.com/brave/brave-core/pull/16351).

Cheers!

**Comment by DesertBear** [comment]
> @DesertBear thanks for requesting that - I logged that one with #29397 and we can look at this soon 😄

While `BraveVPNDisabled` has been added to Brave Group Policy settings, the documentation at the below link is still missing this information.

**Link:**
https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy

**Example:**
| DWord | Accepted Values | Effect |
|--------|--------|--------|
| BraveVPNDisabled | 0 (default)<br>1| 0 = Enabled<br> 1 = Disabled |

**Comment by AlanBreck** [comment]
The support site is actually managed by @Brave-Matt, so I'm looping him in.

**Comment by bsclifton** [comment]
`BraveVPNDisabled` is now added to https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy - thanks @Brave-Matt! 😄👍

**Comment by bsclifton** [comment]
@mherrmann - I'm not sure how much trouble it is to append to the `example/brave.reg` in the policy templates (see https://brave-browser-downloads.s3.brave.com/latest/policy_templates.zip)

But we could hardcode (at least for the moment) the following Brave specific values:
```reg
Windows Registry Editor Version 5.00
; brave version: 124.1.67.42

[HKEY_LOCAL_MACHINE\Software\Policies\BraveSoftware\Brave]
"TorDisabled"=dword:00000001
"IPFSEnabled"=dword:00000000
"BraveRewardsDisabled"=dword:00000001
"BraveWalletDisabled"=dword:00000001
"BraveVPNDisabled"=dword:00000001
"BraveAIChatEnabled"=dword:00000000

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave\BraveShieldsEnabledForUrls]
"1"="[*.]twitter.com"
"2"="https://www.example.com"

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave\BraveShieldsDisabledForUrls]
"1"="https://www.example.com"
"2"="[*.]brave.com"

```

This should cover all the ones listed at https://support.brave.com/hc/en-us/articles/360039248271-Group-Policy#h_01HE8CWCDW9FWDWB74VCGZZEMR

**Comment by mherrmann** [comment]
@bsclifton sure. My previous work on the topic was https://github.com/brave/brave-core/pull/16351, but it became obsolete before it was merged due to upstream changes. The policy templates are a GN target, `brave/components/policy:pack_policy_templates`. Maybe you could extend [pack_policy_templates.py]
(https://github.com/brave/brave-core/blob/3311f572f08cd007bf570e8b21dde09d949548ef/components/policy/pack_policy_templates.py#L31) to append the values you mentioned to brave.reg?

**Comment by Marko-98** [comment]
@bsclifton I'm using these policies to disable Brave Wallet and Brave Rewards from the browser; I have two questions:

1. Is it possible to keep using these policies, but to hide messages "managed by your organization" throughout the browser?
2. When policies are enabled, secure DNS option is disabled automatically. Is there a way to keep secure DNS enable while still managing browser through Group Policy or Registry?

Thank you!

**Comment by DesertBear** [comment]
> 2\. When policies are enabled, secure DNS option is disabled automatically. Is there a way to keep secure DNS enabled while still managing browser through Group Policy or Registry?

This is an issue with all Chromium-based browsers.  You will need to set the below policies to keep DNS-over-HTTPS enabled.

**Controls the mode of DNS-over-HTTPS:**
https://chromeenterprise.google/policies/#DnsOverHttpsMode
- Enabled
- Enable DNS-over-HTTPS without insecure fallback

**Specify URI template of desired DNS-over-HTTPS resolver:**
https://chromeenterprise.google/policies/#DnsOverHttpsTemplates
- Enabled
- `https://security.cloudflare-dns.com/dns-query` or `https://dns.quad9.net/dns-query`

**Comment by Marko-98** [comment]
@DesertBear Thanks! :)

**Comment by mihaiplesa** [comment]
Related https://www.reddit.com/r/brave_browser/comments/1efzjfh/found_a_native_way_to_debloat_brave/

**Comment by bsclifton** [comment]
Making some good progress here 😄

**Comment by LaurenWags** [comment]
Removed `QA Pass-Win64` so we can get a Win 10 check on this one as well 👍🏻

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on

```
Brave | 1.74.25 Chromium: 132.0.6834.33 (Official Build) beta (64-bit)
-- | --
Revision | 041ffab11e000929213dcdae825dba438ff8a620
OS | Windows 10 Version 22H2 (Build 19045.5247)
```

**Comment by GeetaSarvadnya** [comment]
@mherrmann: I am verifying the issue on Windows 10 x64 and Windows 11 x64. I have followed the test plan mentioned via https://github.com/brave/brave-browser/issues/26502#issue-1434352417. On both win10 and win11, the issue is not working as expected.  When the Tor/VPN is enabled via `gpedit.msc` and launch the browser (Brave Beta 1.74.26) is still showing up the VPN via toolbar and NTP etc...and `New private window with Tor` via hamburger menu. 

I have few questions reg the Group Policy template:

1. The policy template support only for the countries provided in the template below or all the countries?
![Image](https://github.com/user-attachments/assets/8b144f75-cf60-4c28-b401-b0929921ba79)
 2. My OS locale was set to default region `India` and language was `English -US` - when it was not working with these combinations, I have changed my system OS region to US and then tried the steps mentioned in the issue description but no luck with this change. 

Unable to disable the Brave-specific browser options via Group Policy settings from India region. The same thing is working fine for @MadhaviSeelam on Windows 11 x64

Please see the recorded file below:

https://drive.google.com/file/d/1mA7c-IUN0eBGukDMow8otXNbwC6d6dbO/view?usp=sharing

**Comment by mherrmann** [comment]
@GeetaSarvadnya the feature was implemented by @bsclifton. I think he will know more about this than me.

**Comment by bsclifton** [comment]
@GeetaSarvadnya the templates are localized for those countries/locales - but usable in any. Unfortunately, none of the Brave strings are being localized. Only the standard Chromium templates are actually localized.  For testing, you can always pick the `en-US` one.

I watched the video - thanks for capturing @GeetaSarvadnya. The file you opened looks correct and you set it properly.

We could look at the registry perhaps as a next step. Specifically, open `regedit` and at the top of the `regedit` tool, you should be able to paste in `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave`.

I'm curious if any values are being shown here. The best way to check policies being set (besides registry) is via brave://policy which you already did check

**Comment by GeetaSarvadnya** [comment]
@bsclifton: When I enable/disable brave specific browser options (e.g VPN/Tor) via `gpedit.msc` local policy editor, there is no registry entry shown via REGEDIT (`Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave`), also when I open brave://policy via browser, no policies are set. 
![Image](https://github.com/user-attachments/assets/d783f7ed-a31f-4760-95e6-38bd8cf85ef3)


But, If I set policies via REGEDIT(Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Policies\BraveSoftware\Brave) path, the policies are working as expected. When I open the browser, the browser options are disabled and also the policies are shown as `true` via brave://policy
![Image](https://github.com/user-attachments/assets/b052f696-4733-48c0-81d8-9de2ce08a9cc)

**Comment by bsclifton** [comment]
@GeetaSarvadnya quick note on the registry locations

For 64 bit Windows, the path should actually be `HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Policies\BraveSoftware\Brave`

Can you check there? I think it's supposed to alias to the same place, but maybe it didn't.

**Comment by bsclifton** [comment]
Also - can you open gpedit.msc as admin? For example, you can open admin command prompt, then do `gpedit.msc` to open it

**Comment by GeetaSarvadnya** [comment]
@bsclifton: Opened `gpedit.msc` via cmd line prompt with Admin rights, set policy via `gpedit.msc` policy editor is not working as expected and I am seeing the same results before. 

![Image](https://github.com/user-attachments/assets/1ec7b746-ac27-497b-9ccf-1eeb260c8c48)

Opened `REGEDIT` with admin rights and set policies via REGEDIT (HKEY_LOCAL_MACHINE\SOFTWARE\WOW6432Node\Policies\BraveSoftware\Brave) works as expected 
![Image](https://github.com/user-attachments/assets/049431a2-492f-48fa-83ef-5ba8f0ecd29d)

**Comment by bsclifton** [comment]
@GeetaSarvadnya what version of Windows are you using?

It seems the regular registry path and the 64-bit path are working in registry. But doesn't seem the group policy enforcement is happening. To be clear, I don't know if the registry keys get created by the policy or not. The registry is just another way to do it. The ultimate test is making sure it shows on brave://policy/ - which you did show is not working (with gpedit.msc)😭 

Are you using the PRO version of Windows 10/11? If not, that might be the problem. I think the easiest way to check is open `Settings` > `System` > `About` and then there's a section called `Windows specifications`. For reference, mine looks like this:

```
Edition	Windows 11 Pro
Version	24H2
Installed on	‎11/‎14/‎2024
OS build	26100.2605
Experience	Windows Feature Experience Pack 1000.26100.36.0
```

**Comment by GeetaSarvadnya** [comment]
@bsclifton: System details are as below:

```
Edition	Windows 10 Home Single Language
Version	22H2
Installed on	‎16-‎11-‎2020
OS build	19045.5247
Experience	Windows Feature Experience Pack 1000.19060.1000.0

```

![Image](https://github.com/user-attachments/assets/e2f347c2-f2eb-4778-ad1e-681e28725d4a)

**Comment by GeetaSarvadnya** [comment]
@bsclifton: I have both Windows 10 and Windows 11 with Home versions. For both versions group policy works as expected via Registry. On both Windows versions group policy editor doesn't work via `gpedit.msc`.

 I don't have Windows 10/11 with Pro version to test the policy editor via `gpedit.msc`. @MadhaviSeelam has already tested the issue on Windows 11 with Pro version and it worked for her. Can we add QA pass label to the issue?

cc: @LaurenWags

**Comment by bsclifton** [comment]
Yup- QA/Pass SGTM! Thanks 😄

**Comment by kjozwiak** [comment]
> [@GeetaSarvadnya](https://github.com/GeetaSarvadnya) OK great - I think we have the smoking gun 😄
> 
> * Windows Home version can do group policy via Registry
> * Windows Pro version is required to enforce group policy
> 
> When using Active Directory / setting up domain controllers, you need the pro version to join the domain. So I think it's OK - most people looking at `gpedit.msc` are going to be Pro customers. Home version is not intended to handle group policy or join a work domain.

Thanks @bsclifton 👍 Definitely the case. I quickly asked Leo the same thing before catching up on the issue and got the same answer/results re: `Home` edition not being able to use/set `Policies` with `gpedit.msc`.

![Image](https://github.com/user-attachments/assets/b4ae432e-00fd-4f11-8970-acab1d0028bb)

Fortunately my `Win 10 x86` machine is a `Pro` version so I was able to check/run through the above as per the following:

![Image](https://github.com/user-attachments/assets/0650868c-927a-48b9-9391-f604d078f391)

Verification **PASSED** on `Win 10 x86 Pro` using the following build(s):

```
Brave | 1.74.46 Chromium: 132.0.6834.83 (Official Build) (32-bit)
-- | --
Revision | e4202daaaf217419a45c550e19acb7d7e413dda1
OS | Windows 10 Version 22H2 (Build 19045.5011)
```

Using the STR/Cases outlined via https://github.com/brave/brave-browser/issues/26502#issue-1434352417, ensured that setting `Brave` group policies via `User Configurations` -> `Classic Administrative Template (ADM)` -> `Brave` -> `Brave` -> `Brave Software settings` were being applied without any issues as per the following:

**`Example`** | **`Example`** | **`Example`**
----------------|----------------|-----------------
![Image](https://github.com/user-attachments/assets/c2e8266f-91b3-4076-b9c7-acbed1e11b2f) | ![Image](https://github.com/user-attachments/assets/d107f2bd-1461-4c83-b6b2-6943b7fb3351) | ![Image](https://github.com/user-attachments/assets/b43f9cc3-40f5-4fb5-9831-823e5291d601)

**Comment by LaurenWags** [comment]
Updated to `release-notes/exclude` per discussion with @bsclifton
